Userland/Applications/PixelPaint/Tools/GradientTool.cpp

```diff
@@ -346,32 +403,72 @@ void GradientTool::reset()
     m_gradient_start = {};
     m_gradient_center = {};
     m_gradient_end = {};
+    m_gradient_left = {};
+    m_gradient_right = {};
     m_gradient_half_length = 0;
     m_physical_diagonal_layer_length = 0;
     m_hover_over_drag_handle = false;
     m_hover_over_start_handle = false;
     m_hover_over_end_handle = false;
+    m_hover_over_left_handle = false;
+    m_hover_over_right_handle = false;
 
     if (m_editor) {
         m_editor->update();
         m_editor->update_tool_cursor();
     }
 }
 
-void GradientTool::update_gradient_end_and_derive_start(Gfx::IntPoint const new_end_point)
+void GradientTool::update_gradient_with_initial_values(Gfx::IntPoint const new_end_point)
 {
     VERIFY(m_gradient_center.has_value());
     m_gradient_end = new_end_point;
-    m_gradient_start = m_gradient_center.value() - (m_gradient_end.value() - m_gradient_center.value());
+    auto deltaCenter = m_gradient_end.value() - m_gradient_center.value();
+    m_gradient_start = m_gradient_center.value() - deltaCenter;
+
+    if (m_mode == GradientMode::Radial) {
+        // rotate deltaCenter by 90 degrees to derive the left and right point
+        deltaCenter = { -deltaCenter.y(), deltaCenter.x() };
```

Comment:
thatʼs delta center no more; you could avoid the comment by setting new variable with a better name (and have the code commenting itself properly 😉).